test/test_utils.h

```diff
@@ -177,15 +179,19 @@ namespace test_utils {
 		cdag_inspector& get_inspector() { return inspector; }
 		detail::graph_serializer& get_graph_serializer() { return *gsrlzr; }
 
-		void build_task_horizons() {
+		detail::task_id build_task_horizons() {
 			auto most_recently_generated_task_horizon = detail::task_manager_testspy::get_current_horizon_task(get_task_manager());
 			if(most_recently_generated_task_horizon != most_recently_built_task_horizon) {
 				most_recently_built_task_horizon = most_recently_generated_task_horizon;
 				if(most_recently_built_task_horizon != nullptr) {
-					auto htid = most_recently_built_task_horizon->get_id();
-					get_graph_generator().build_task(htid, {nullptr});
+					const auto htid = most_recently_built_task_horizon->get_id();
+					// naive_split does not really do anything for horizons, but this mirrors the behavior of scheduler::schedule exactly.
+					detail::naive_split_transformer naive_split(num_nodes, num_nodes);
+					get_graph_generator().build_task(htid, {&naive_split});
+					return htid;
```

Comment:
As I mentioned during the meeting, I think having to reproduce a lot of the logic in the test utils is not ideal. But I also don't have a better idea for unit tests which shouldn't have the entire runtime in the background.

Reply:
I think we could reasonably manage the set of active transformers in `graph_generator` and get rid of the `native_split` construction (and vector allocation) here. Thoughts @psalz ?

Reply:
Hmm, I'm not sure. While the scheduler is pretty bare-bones at the moment, I have at various occasions experimented with multiple/different transformers depending on task types, hints etc -- which is something that I don't think the graph generator should be aware of. We can certainly revisit the current scheduler/transformer/ggen design, but I'd say its out of scope for this PR.

Reply:
Let's just keep it in mind then.